overlord/snapstate/snapstate.go

```diff
@@ -1358,6 +1358,117 @@ func installWithDeviceContext(st *state.State, name string, opts *RevisionOption
 	return doInstall(st, &snapst, snapsup, 0, fromChange, nil)
 }
 
+type snapInfoForDownload func(DeviceContext, *RevisionOptions) (si *snap.Info, redirectChannel string, e error)
+
+func Download(ctx context.Context, st *state.State, name string, opts *RevisionOptions, userID int, flags Flags, deviceCtx DeviceContext, fromChange string) (*state.TaskSet, error) {
+	snapDownloadInfo := func(dc DeviceContext, ro *RevisionOptions) (si *snap.Info, redirectChannel string, e error) {
```

Comment:
I'm not sure if this indirection/abstraction is needed, but all of the other functions like this follow a similar pattern.

Reply:
no, it's a bit overkill, unless we plan to call download() from many places. InstallPath has doesn't use such abstraction fwiw